src/solvers/mp/rcspp/RCGraph.cpp

```diff
@@ -0,0 +1,1095 @@
+//
+// Created by antoine legrain on 2020-04-10.
+//
+
+#include "solvers/mp/rcspp/RCGraph.h"
+#include <iostream>
+
+// print a solution
+std::string RCSolution::toString(std::vector<int> shiftIDToShiftTypeID) const {
+  std::stringstream buff;
+  buff << "RC solution of cost " << cost
+       << " starting on day " << firstDay << ":" << std::endl;
+  for(int k=0; k<firstDay; k++) buff << "|     ";
+
+  char buffer[500];
+  if(shiftIDToShiftTypeID.empty())
+    for(int s: shifts) {
+      std::sprintf(buffer,  "| -:%2d", s);
+      buff << buffer;
+    }
+  else
+    for(int s: shifts) {
+      std::sprintf(buffer, "|%2d:%2d", shiftIDToShiftTypeID[s], s);
+      buff << buffer;
+    }
+  buff <<  buffer  <<  "|" << std::endl;
+  return buff.str();
+}
+
+// Resources extension model (arc has cost + label consumptions)
+    bool ref_spp::operator()( const Graph& g,
+                            spp_res_cont& new_cont,
+                            const spp_res_cont& old_cont,
+                            boost::graph_traits<Graph>::edge_descriptor ed ) const {
+  const Arc_Properties &arc_prop = get(boost::edge_bundle, g)[ed];
+  if (arc_prop.forbidden)
+    return false;
+  const Vertex_Properties &vert_prop = get(boost::vertex_bundle, g)[target(ed, g)];
+  if (vert_prop.forbidden)
+    return false;
+  new_cont.cost = old_cont.cost + arc_prop.cost;
+
+  for (int l = 0; l < old_cont.size(); ++l) {
+    int lv = std::max(vert_prop.lb(l), old_cont.label_value(l) + arc_prop.consumption(l));
```

Comment:
Here, the update of lv seems to be a reference to time windows. Since we don't have those, maybe revise this line to avoid any trouble.

Reply:
I have stated to do it in the new refactoring  I'm working on, and I thought we could decide that every bounds are hard (or I have added a hard field in the vertex).